src/fileActions.jsx

```diff
@@ -526,6 +526,22 @@ const setDirectoryName = (val, setName, setNameError, setErrorMessage) => {
     }
 };
 
+const downloadFile = (currentPath, selected) => {
+    const query = window.btoa(JSON.stringify({
+        payload: "fsread1",
+        binary: "raw",
+        path: `${currentPath}/${selected.name}`,
+        superuser: "try",
+        external: {
+            "content-disposition": `attachment; filename="${selected.name}"`,
+            "content-type": "application/octet-stream",
+        }
+    }));
+
+    const prefix = (new URL(cockpit.transport.uri("channel/" + cockpit.transport.csrf_token))).pathname;
+    window.open(`${prefix}?${query}`, '_blank');
```

Comment:
I think we decided that the `_blank` thing is a no-go.  @garrett didn't like the flicker and pop-up blockers didn't like it either.